### ISSUE TITLE
reward_fn evaluation time too long, gpu empty running

### ISSUE THREAD

**Issue by RTkenny** [wizard-III/WizardCodeR#6]
Dear author
Thank you for your very interesting work. When I am running your experiment on code task. The time for reward_fn is very long(more than 1000 second), which mean that the cpu is executing the generated code to produce the reward score. I understand that to evaluate code is more time-comsuming than math problems. However, the executing time for the code's reward_fn is too long in my experiment. I have check your implementation for the reward_fn in the training set, there is no timeout limit for the code execution. I wonder how you handles this problems in your experiments. Since overlong reward_fn evaluation make the gpu utility near 0 and wasting a lot time, , which is unacceptable. I sincerely looking forward to your reply.

Best regard

<img width="1060" height="779" alt="Image" src="https://github.com/user-attachments/assets/7a2c5779-59ef-434e-8a6d-edd986dd761c" />

**Comment by wizard-III** [comment]
Thanks for your interest. Regarding the issue of code reward being time-consuming:

First, during training data processing, we filtered out some overly long tail test cases.

Second, we also controlled the number of test cases for each problem in train set.

Finally, there is actually a timeout limit set in the code reward ~ https://github.com/wizard-III/ArcherCodeR/blob/main/rewards/code_utils/utils.py#L32

**Comment by RTkenny** [comment]
Thank you for your detailed answer!
